### ISSUE TITLE
Respect DD_AGENT_HOST environment variable when setting the default agent endpoint.

### ISSUE THREAD

**Issue by twelvelabs** [DataDog/kong-plugin-ddtrace#15]
I am using the [Datadog Admission Controller](https://docs.datadoghq.com/containers/cluster_agent/admission_controller) in a k8s cluster. The controller injects the `DD_AGENT_HOST` env variable into pods at startup.

I tried using the following in my declarative config:

```yaml
agent_endpoint: 'http://{vault://env/dd-agent-host}:8126/v0.4/traces'
```

... but it appears that the secrets management logic does not support nesting `vault:` placeholders in another string - it only replaces them when the string begins and ends with the placeholder.

I'm currently using `envsubst` to swap out `$DD_AGENT_HOST` in the kong.yaml file in the Docker entrypoint, but it would great if the plugin could just use that value in the default if present.

**Comment by twelvelabs** [comment]
No, I wouldn't expect the plugin to parse the config string differently. More the later - ideally the plugin would inspect the environment when setting the default. IIRC, that's the default behavior in some of the other ddtrace implementations I've used.

**Comment by dgoffredo** [comment]
Inspecting the environment from within an nginx worker processes can be a problem, because nginx does not forward most environment variables to its children. They must be explicitly named with [env][1] in `nginx.conf`.

I'll bring this up with my teammates.

An alternative would be to have an option:
```yaml
agent_host: '{vault://env/dd-agent-host}'
```
if we don't want to open up the environ can of worms.

[1]: http://nginx.org/en/docs/ngx_core_module.html#env

**Comment by dmehala** [comment]
Hello @twelvelabs,

Appreciate your report on this issue. I'd like to update you that we've successfully merged the solution, and it will be part of the forthcoming release.

Thank you for bringing this to our attention. If you have any additional questions or concerns, feel free to reach out.
